bedwars-plugin/src/main/java/com/tomkeuper/bedwars/addon/AddonManager.java

```diff
@@ -2,44 +2,85 @@
 
 import com.tomkeuper.bedwars.BedWars;
 import com.tomkeuper.bedwars.api.addon.Addon;
+import com.tomkeuper.bedwars.api.addon.AddonStorer;
 import com.tomkeuper.bedwars.api.addon.IAddonManager;
-import java.util.ArrayList;
+import org.bukkit.Bukkit;
+
 import java.util.List;
 import java.util.stream.Collectors;
 
-public class AddonManager implements IAddonManager {
-    private static List<Addon> addons;
+public class AddonManager extends AddonStorer implements IAddonManager {
+    private static List<Addon> registeredAddons;
+    private static List<Addon> unloadedAddons;
+    private static List<Addon> loadedAddons;
```

Comment:
You define the lists here but also within the  addonStorer class, whats the point of the AddonStorer within the API?

Reply:
i dont want to add the AddonManager into the API package so i just make a "tunnel" to reach the AddonManager class

Reply:
If you know any way better than this one please tell me

Reply:
If you add the functions to the bw plugin API class you can refrence to the BedWars class and get the addon manager from there. That way you can also return the values

Reply:
@KKiiya if you need me to elaborate just let me know

Reply:
What do you mean, didnt understand

Reply:
Instead of making an AddonStorer class that is extended bt this class you should use the API class within the plugin to publish these values.  
Currently there is a API#getAddonsUtil() function that returns the addonManager class, with this class you can already get all the public functions that are avaiable in this AddonManager class. 

The addon storer doesnt really do anything it looks like as all values are also avaiable using the IAddonManager interface

Reply:
I see, if you check the Addon class, you will see that is used there to add the registered addons in that class. But now that you've said that, i realized that the loaded addons and unloaded addons lists dont do anything there

Reply:
the registeredAddons variable is also defined in the addonManager class so the whole AddedStorer class is not needed.